### PR TITLE
Foundation for Shipping Addresses

### PR DESCRIPTION
**Pull request by DanielVF**
This PR lays the foundation for encrypted shipping addresses sent as a part of an offer.

## Messaging Layer

Added the concept of an OutOfBandMessage that can be sent separately from the normal messaging flow, but can be encrypted and decrypted using the same mechanisms as regular ordered conversations. 

Add a few tests to messaging-client to verify that regular messages, and out of band messages, can be encrypted and decrypted. This mocks out the the browser and uses the production messaging server.

Added a tiny example docstring at the top of the messaging client to help someone else get started playing with it.

## GraphQL Layer

Added a `shippingAddress: string` field to the makeOffer mutation. The contents are encrypted and included in the offer. If passed in, and both parties do not have messaging enabled, this throws an error. 

Added a `shippingAddressEncrypted` field to offers. This can be decrypted with the messaging `messaging.decryptOutOfBandMessage` query.

## DAPP Layer

I've put a placeholder shipping address display support into the DAPP. The shipping address shows up on the transaction page if a shipping address string is included.

I've not included user facing to send the shipping address - just set the `shippingAddress` variable of the makeOffer mutation.

## Deployment

After review, this should be safe to merge. Nothing user facing in the DAPP should change.

@tomlinton This adds tests in packages/messaging-client. We could add these to CI.

### Checklist:

- [x] Test your work and double-check to confirm that you didn't break anything
- [x] [Wrap any new text/strings with `fbt`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#wrapping-text) for translation
- [x] If there are any new/changed strings to translate, [`npm run translate`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#integrating-translations)

### CODE REVIEW

File: dapps/marketplace/src/queries/DecryptOutOfBandMessage.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+const DecryptOutOfBandMessage = gql`
+  query DecryptOutOfBandMessage($encrypted: String!) {
+    messaging(id: "defaultAccount") {
+      decryptOutOfBandMessage(encrypted: $encrypted) {
+        content
+      }
+    }
+  }
+`
+window.DecryptOutOfBandMessage = DecryptOutOfBandMessage
```
[the remark this answers is not in the document]
Comment (DanielVF) [reply dapps/marketplace/src/queries/DecryptOutOfBandMessage.js]: Removed, thanks!

File: packages/messaging-client/src/Messaging.js [hunk 2]
```diff
@@ -786,6 +846,54 @@ class Messaging {
...
+      }
+      return this.web3.utils.toChecksumAddress(x)
+    })
+    // Sort my address to the end of the array
+    // Making the the first address not me (unless I'm messaging myself)
+    addresses.sort(x => (x == this.account_key ? 1 : -1))
+    const remoteEthAddress = addresses[0]
+
+    const convObj = await this.startConv(remoteEthAddress)
+    console.log('Conversation object', convObj)
```
[the remark this answers is not in the document]
Comment (DanielVF) [reply packages/messaging-client/src/Messaging.js]: Removing

File: packages/messaging-client/src/Messaging.js [hunk 3]
```diff
@@ -786,6 +846,54 @@ class Messaging {
...
+    const addresses = [message.address, message.to].map(x => {
+      if (!this.web3.utils.isAddress(x)) {
+        throw new Error(`${x} is not a valid Ethereum address`)
+      }
+      return this.web3.utils.toChecksumAddress(x)
+    })
+    // Sort my address to the end of the array
+    // Making the the first address not me (unless I'm messaging myself)
+    addresses.sort(x => (x == this.account_key ? 1 : -1))
+    const remoteEthAddress = addresses[0]
```
[the remark this answers is not in the document]
Comment (DanielVF) [reply packages/messaging-client/src/Messaging.js]: Maybe I'm over thinking it, but I wanted to support a message from yourself, to yourself, in which case a filter would not work.

[the remark this answers is not in the document]
Comment (franckc) [reply packages/messaging-client/src/Messaging.js]: Got it. Sounds good !

**Comment by franckc** [comment]
@shahthepro FYI this is the shipping address back-end PR. There is a unit test failing that @DanielVF will look into and then this will get merged to master.

**Pull request merged**